src/plugins/launch/slurm/launch_slurm.c

```diff
@@ -590,6 +590,53 @@ extern int launch_p_create_job_step(srun_job_t *job, bool use_all_cpus,
 	return SLURM_SUCCESS;
 }
 
+static char *_find_quote_token(char *tmp, char *sep, char **last)
```

Comment:
This should just be `strsep`. Drop the function, it's not needed.

Reply:
it is. try to split `X,Y=1,2,Z` properly without custom quoting. 

anyway, complain to schedmd that their code is bad.

Reply:
Ah, I think I see the issue, you need something to deal with `'1,2',3,4`, right?

Reply:
yes. the true issue is that schedmd uses different code to parse the sbatch and srun --export option value.

Reply:
Also, you only use `sep[0]`, so ideally, this is passed as a char, not as a char*.

Reply:
I do not like the below code, fiddling with the `i` inside the loop. Better would be to use 

~~~~
char* split(char *buffer, const char sep, char **last)
{
    char *p, *start_of_word;
    int c;
    enum states { DULL, IN_WORD, IN_STRING_D, IN_STRING_S } state = DULL;

    char *start = (buffer == NULL ? *last : buffer);

    if (start == NULL) {
        return NULL;
    }

    for (p = start; *p != '\0'; p++) {
        c = (unsigned char) *p;
        switch (state) {
        case DULL:
            if (c == sep) {
                continue;
            }

            if (c == '"') {
                state = IN_STRING_D;
                if (*(p+1) == '\0') {
                    return NULL;
                }
                start_of_word = p + 1;
                continue;
            }

            if (c == '\'') {
                state = IN_STRING_S;
                if (*(p+1) == '\0') {
                    return NULL;
                }
                start_of_word = p + 1;
                continue;
            }

            state = IN_WORD;
            start_of_word = p;
            continue;

        case IN_STRING_D:
            if (c == '"') {
                *p = 0;
                *last = (*(p+1) != '\0') ? p+1 : NULL;
                return start_of_word;
            }
            continue;

        case IN_STRING_S:
            if (c == '\'') {
                *p = 0;
                *last = (*(p+1) != '\0') ? p+1 : NULL;
                return start_of_word;
            }
            continue;

        case IN_WORD:
            if (c == sep) {
                *p = 0;
                *last = (*(p+1) != '\0') ? p+1 : NULL;
                return start_of_word;
            }
            continue;
        }
    }

    if (state != DULL) {
        *last = NULL;
        return start_of_word;
    }

    return NULL;
}
~~~~

Based on https://stackoverflow.com/a/26913698/349809. If we ever need more bells and whistles it is easy to add.

Reply:
not sure how manytimes i have to repeat this, but schedmd will fix the issue, and then this commit can be reverted.

Reply:
I had not heard that, holidays and all that, so maybe one more time :p

Reply:
Did you open a ticket with them for it? If so, which one?

Reply:
https://bugs.schedmd.com/show_bug.cgi?id=7658